lib/src/mount.rs

```diff
@@ -57,3 +57,19 @@ pub(crate) fn mount(dev: &str, target: &Utf8Path) -> Result<()> {
         [dev, target.as_str()],
     )
 }
+
+// Ensure the `/dev` directory exists.
+fn ensure_mountpoint(target: &str) -> Result<()> {
+    std::fs::create_dir_all(target)?;
+    Ok(())
+}
+
+/// General mount that also checks that mount path exists
+pub(crate) fn ensure_mount(dev: &str, target: &str, fstype: &str) -> Result<()> {
```

Comment:
Minor/optional: Function with especially >2 parameters all of the same type can be easy to call with the wrong arguments.  We could change `fstype` to be an `enum FilesystemType { Devtmpfs }` for example.

Another thing to do (which can be combined) is to use `&Utf8Path` here instead of `&str` which expresses semantically that it's a path.

Reply:
I went to look at [the rustix mount API](https://docs.rs/rustix/latest/rustix/mount/fn.mount.html) and saw a "Fs" parameter which I thought would be an enum like this, but it's actually just a generic bytestring (ultimately of course we don't want to *need* to change rustix/glibc when a new Linux filesystem appears so I guess it has to be just a string, but in this code nothing stops us from having an enum, feels like it would make sense)